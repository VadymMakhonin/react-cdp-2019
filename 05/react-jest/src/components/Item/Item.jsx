import React from 'react';
import { dateFormater } from './../../utils/common';


export const Item = ({ todo, onRemove, onChange }) => (
    <div className="row" key={todo.id}>
        <div className="col-1">
            <input type="checkbox" checked={todo.complited} onChange={(event) => {
                onChange(todo.id, event.target.checked);
            }} />
        </div>
        <div className="col-1">{todo.id}</div>
        <div className="col-7">{todo.title}</div>
        <div className="col-1">{dateFormater(todo.created)}</div>
        <div className="col-2 text-right">
            <button className="btn btn-danger" onClick={() => { onRemove(todo.id) }}>
                Remove
                </button>
        </div>

    </div>
);