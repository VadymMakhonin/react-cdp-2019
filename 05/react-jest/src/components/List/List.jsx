import React from 'react';
import {Item} from './../Item/Item';

export const List = ({ todos, onRemove, onChange }) => (todos.length ? todos.map((todo) => (
    <Item todo={todo} onRemove={onRemove} onChange={onChange}/>
)) : 'Todos is empty');