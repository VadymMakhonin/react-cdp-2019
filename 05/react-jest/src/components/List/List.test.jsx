import React from 'react';
import {List} from './List';
import {Item} from './../Item/Item';
import {shallow, mount} from 'enzyme';

describe('List component', () => {

    it('should be render text "Todos is empty"', () => {
        const todos = [];
        const component = shallow(<List todos={todos} />);
        expect(component).toMatchSnapshot();
    });

    it('should be render two elements', () => {
        const todos = [{title: 'test'}, {title: 'test 3'}];
        const component = shallow(<List todos={todos} />);
        expect(component).toMatchSnapshot();
    });

    it('should be render two component - Item', () => {
        const todos = [{title: 'test', id: Math.random()}, {title: 'test 2', id: Math.random()}];
        const component = mount(<List todos={todos} />);
        expect(component.find(Item).length).toBe(2);
    });
})