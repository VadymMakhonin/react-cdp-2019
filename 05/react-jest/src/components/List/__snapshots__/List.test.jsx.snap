// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should be render text "Todos is empty" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    todos={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "Todos is empty",
  Symbol(enzyme.__nodes__): Array [
    "Todos is empty",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`List component should be render two elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    todos={
      Array [
        Object {
          "title": "test",
        },
        Object {
          "title": "test 2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "onChange": undefined,
      "onRemove": undefined,
      "todo": Object {
        "title": "test",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onChange": undefined,
        "onRemove": undefined,
        "todo": Object {
          "title": "test",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onChange": undefined,
        "onRemove": undefined,
        "todo": Object {
          "title": "test 2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
